site/en/docs/extensions/mv3/content_scripts/index.md

```diff
@@ -171,41 +179,54 @@ They can include JavaScript files, CSS files, or both. All auto-run content scri
   </tbody>
 </table>
 
-{% if false %}
 ### Inject with dynamic declarations {: #dynamic-declarative }
 
-{% Aside 'caution' %}
-
-This feature is not yet fully supported. It is currently in dev and is also available in Chrome
-Canary.
-
-{% endAside %}
-
-You should use dynamic declarations in the following cases:
+Dynamic content scripts are useful for situations when the match patterns for a content scripts are
+not well known or when content scripts should not always be injected on known hosts.
 
-- When the host is not well known
-- The script may need to be added/removed from a known host
+Introduced in Chrome 96, dynamic declarations are similar to [static declarations][14], but the
+content script object is registered with Chrome using methods on the Scripting API rather than in
+[manifest.json][doc-manifest]. In addition to [registering][api-register-cs] content scripts, the
+Scripting API also allows extension developers to [get a list of][api-get-registered-cs],
+[update][api-update-cs], and [remove][api-remove-cs] content script registrations.
 
-**TODO**
+Like static declarations, dynamic declarations can include JavaScript files, CSS files, or both.
+Unlike static declarations, dynamic content scripts cannot be [injected into related
+frames][header-related-frames].
 
-- Uses the JS scripting API
-    - example of adding a script
-    - example of removing a script
-- Q: Is this going to be the cannonical reference material for content scripts? If so, we may want
-  to include examples for the relevant Scripting API methods.
+```js
+chrome.scripting
+  .registerContentScripts([{
+    id: 'session-script',
+    js: ['content.js'],
+    persistAcrossSessions: false,
+    matches: ['*://example.com/*'],
+    runAt: 'document_start',
+  }])
+  .then(() => console.log('registration complete'))
+  .catch((err) => console.warn('unexpected error', err))
+```
```

Comment:
Putting these code samples here makes it unclear if I should read this page or the scripting API page for this API.  Would it be better to just leave this page as "Content scripts can be dynamically registered through the [Scripting API]()", and put all the content there?

(I'm open to alternatives, too, but right now it feels a little fragmented)

Reply:
I see your point. I'll experiment with moving things around soon.

Reply:
I wasn't wild about anything I tried. If you don't mind, I'd like to land this as is and work with/defer to @AmySteam to help clean this up.

Reply:
I agree with @rdcronin.

Reply:
> I wasn't wild about anything I tried. If you don't mind, I'd like to land this as is and work with/defer to @AmySteam to help clean this up.

@dotproto @jpmedley Happy to help tweak the structure and content on a separate PR 👍

Reply:
Please do.